xiilib/_gunicorn/secret_storage.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the SecretStorage for managing the persistent secret storage for the charm."""
+import secrets
+
+import ops
+
+import xiilib.secret_storage
+
+
+class GunicornSecretStorage(xiilib.secret_storage.SecretStorage):
+    """A class that manages secret keys required by the WSGI charms."""
+
+    def gen_initial_value(self) -> dict[str, str]:
+        """Generate the initial secret values.
+
+        Returns:
+            The initial secret values.
+        """
+        return {self._key: secrets.token_urlsafe(32)}
+
+    def __init__(self, charm: ops.CharmBase, key: str):
+        """Initialize the SecretStorage with a given charm object.
+
+        Args:
+            charm: The charm object that uses the SecretStorage.
+            key: The secret key name stored in the relation data.
+        """
+        super().__init__(charm=charm, keys=[key])
+        self._charm = charm
```

Comment:
Isn't it already set by the parent class ?

Reply:
Indeed, I will remove this, thanks!

Reply:
Removed, thanks!